### PR TITLE
Lint changes for 2021.1

### PR DESCRIPTION
**Pull request by seanbudd**
### Link to issue number:

None

### Summary of the issue:

- various code-fences are not formatted properly
- inconsistent indentation on lists
- inconsistent formatting of messages

### Description of how this pull request fixes the issue:

- enforce `- Example changelog message. (#123[, #321])` format using the following regex to highlight conforming messages `(.*\. \((#[0-9]+(, )?)+\)$|  -.*)`
- replace all usages of single ` with ``
- standarise indentation for second level lists to use the recommend two spaces

### Testing strategy:

None

### Known issues with pull request:

None

### Code Review Checklist:

- [x] Pull Request description is up to date.
- [x] Unit tests.
- [x] System (end to end) tests.
- [x] Manual tests.
- [x] User Documentation.
- [x] Change log entry.
- [x] Context sensitive help for GUI changes.

### CODE REVIEW

**Comment by feerrenrut** [comment]
Testing strategy should include using a smart diff tool (like beyond compare) to diff the generated HTML for the changes file before and after.

File: user_docs/en/changes.t2t [hunk 1]
```diff
@@ -926,7 +926,7 @@ Highlights of this release include Support for tables in Kindle for PC, support
 
 
 = 2018.1.1 =
-This is a special release of NVDA which addresses   a bug in the Onecore Windows Speech synthesizer driver, which was causing it to speak with a higher pitch and speed in Windows 10 Redstone 4 (1803). (#8082)  
+This is a special release of NVDA which addresses  a bug in the Onecore Windows Speech synthesizer driver, which was causing it to speak with a higher pitch and speed in Windows 10 Redstone 4 (1803). (#8082)
```
Comment (feerrenrut) [inline user_docs/en/changes.t2t:929]: This removed a space, but there is still a double space here. IE it was a triple space.

File: user_docs/en/changes.t2t [hunk 2]
```diff
@@ -1103,7 +1103,7 @@ Highlights of this release include input of contracted braille, support for new
 == Bug Fixes ==
 - Editable div elements in Chrome no longer have their label reported as their value while in browse mode. (#7153)
 - Pressing end while in browse mode for an empty Microsoft Word document no longer causes a runtime error. (#7009)
-- Browse mode is now correctly   supported in Microsoft Edge where a document has been given a specific ARIA role of document. (#6998)
+- Browse mode is now correctly  supported in Microsoft Edge where a document has been given a specific ARIA role of document. (#6998)
```
Comment (feerrenrut) [inline user_docs/en/changes.t2t:1106]: Also still double space

**Review by feerrenrut — COMMENTED** [review]
This is might be hard for translators to work with. All these lines will show up as changed (at least for changelog prior to 2021.1), You might need to couple this with an explanation on the translators mailing list.

I've skimmed through the changes, it is easiest to look at the character diff (for me at least). It might be easier for others to review this if the changes were grouped into commits by type of change. Eg all of the single backtick to double back tick changes in a single commit, all the double space to single space in another commit. 

From the PR description it seems like it may have been unintentional to update the spaces mid-sentence?

**Comment by XLTechie** [comment]
@seanbudd 
>     * inconsistent indentation on lists

As I pointed out in PR #12412. However, I must be missing something about lists. You say:

>     * replace indentation with more than 1 space with 1 space

[This page on txt2tags.org](https://txt2tags.org/markup.html) suggests two spaces for list indentation. Why are we preferring one?
I admit to being a newcomer to T2T--I'm much more familiar with Markdown--but if the homepage of the language says two spaces, wouldn't most people expect that to be correct?

Speaking of #12412, It most likely conflicts with this. I fired it before seeing this. I apologize for the crossover.

**Comment by seanbudd** [comment]
@feerrenrut 

> This is might be hard for translators to work with. All these lines will show up as changed (at least for changelog prior to 2021.1), You might need to couple this with an explanation on the translators mailing list.

Would it be better to leave changes prior to 2021.1 unlinted? I can write something up otherwise. 

>I've skimmed through the changes, it is easiest to look at the character diff (for me at least). It might be easier for others to review this if the changes were grouped into commits by type of change. Eg all of the single backtick to double back tick changes in a single commit, all the double space to single space in another commit.

Yes can do, that makes sense.

>From the PR description it seems like it may have been unintentional to update the spaces mid-sentence?

I think so, should these be linted as well or left alone? I don't think they matter much, I think I tried to fix a few manually on lines that were already linted for other reasons. 

@XLTechie 

>I admit to being a newcomer to T2T--I'm much more familiar with Markdown--but if the website of the language says two spaces, wouldn't most people expect that to be correct?
I looked in the Developer Guide and wiki, and found nothing on this.

I think it was a more minimal change than using 2 consistently, but if we are to lint the whole file and warn translators we should probably use 2. 

>Speaking of #12412, It most likely conflicts with this PR. I fired it before seeing this. I apologize for the crossover. I or someone may close it if those changes aren't thought worth making.

I think that's fine, it's good to have multiple reviews of the changes file. We can resolve whatever conflicts if they arise but the PR seems worth keeping open until it is merged.

**Comment by feerrenrut** [comment]
Can we just do the 2021.1 section modifications now, and follow up another PR for the older sections. So we can merge translations to beta.

**Review by feerrenrut — APPROVED** [review]
Thanks @seanbudd

**Pull request merged**